CMake Install fails on 22.04-based system

**What version of protobuf and what language are you using?**
v25.1

For use in C++, but no relevant here

**What operating system (Linux, Windows, ...) and version?**
Debian 22.04-based docker image (`buildpack-deps:22.04`)

**What runtime / compiler are you using (e.g., python version or gcc version)**
g++ 11.4.0

**What did you do?**
build this docker image:

```
FROM buildpack-deps:22.04

RUN apt clean -y && apt update -y && apt install -y ca-certificates
ARG DEBIAN_FRONTEND=noninteractive
ENV TZ=Etc/UTC
RUN apt-get install -y tzdata
RUN apt-get update --fix-missing

RUN apt-get install -y wget cmake gcc git less vim
WORKDIR /
RUN git clone --branch v25.1 https://github.com/protocolbuffers/protobuf
WORKDIR /protobuf
RUN git submodule update --init --recursive
RUN mkdir build
WORKDIR /protobuf/build
RUN cmake -Dprotobuf_BUILD_TESTS=OFF .. && make -j && make install
```

**What did you expect to see**
Successful install

**What did you see instead?**
`make install` fails at the very end with the following:
```
CMake Error at cmake_install.cmake:1040 (file):                                                                                                                                          
  file INSTALL cannot find                                                                                                                                                               
  "/protobuf/java/core/src/main/java/com/google/java_features.proto": No such                                                                                                            
  file or directory.                                                                                                                                                                     
                                                                                                                                                                                         
                                                                                                                                                                                         
make: *** [Makefile:100: install] Error 1         
```

--- a/cmake/install.cmake
+++ b/cmake/install.cmake
@@ -78,7 +78,10 @@ foreach(_header ${protobuf_HEADERS})
   elseif (_find_nosrc GREATER -1)
     set(_from_dir "${protobuf_SOURCE_DIR}")
   endif()
-  string(REPLACE "${_from_dir}" "" _header ${_header})
+  # On some platforms `_form_dir` ends up being just "protobuf", which can
+  # easily match multiple times in our paths.  We force it to only replace
+  # prefixes to avoid this case.
+  string(REGEX REPLACE "^${_from_dir}" "" _header ${_header})
   get_filename_component(_extract_from "${_from_dir}/${_header}" ABSOLUTE)
   get_filename_component(_extract_name ${_header} NAME)
   get_filename_component(_extract_to "${CMAKE_INSTALL_INCLUDEDIR}/${_header}" DIRECTORY)

